sourcegraph/cody
# Autocomplete: Connect to Fireworks via Sourcegraph Server

This PR changes the `unstable-fireworks` provider to no longer directly connect to Fireworks but instead use the updated code completion endpoint from https://github.com/sourcegraph/sourcegraph/pull/56120.

Note 1: These API calls will fail if SG is not run in dotcom mode.

Note 2: This will also mean that `unstable-firworks` is going to no longer work for dogfooders (I assume this affects very few people though)

Note 3: This will start using streaming the same way that anthropic does

## Test plan

- Set up the Cody Gateway with the `CODY_GATEWAY_FIREWORKS_ACCESS_TOKEN` on localhost
- Start `sg start dotcom` in the sg repo
- Connect Cody to the local instance
- Change the provider to be `unstable-fireworks`
- Observe requests made to the right backend

```
█ CompletionLogger:onComplete: {"type":"completion","status":"success","duration":574}  {
  "result": {
    "completion": "'Hello world');",
    "stopReason": "stop"
  },
  "params": {
    "messages": [
      {
        "speaker": "human",
        "text": "<fim_prefix>// Path: test.ts\nconsole.log(<fim_suffix>\n<fim_middle>"
      }
    ],
    "maxTokensToSample": 256,
    "temperature": 0.4,
    "topP": 0.95,
    "model": "fireworks/accounts/fireworks/models/starcoder-7b-w8a16"
  }
}
```
